src/component/public.ts

```diff
@@ -163,6 +164,70 @@ export const listRoom = query({
   },
 });
 
+export const listAllUsers = query({
```

Comment:
A bit concerned about exposing this directly via the API since it'll likely hit limits pretty quickly for many applications. One alternative would be to paginate the responses from this. The other would be to just have a `listRooms` function that lists the rooms and then these can be used with `listRoom` to get the users in each. That would keep cross-room operations to a minimum and minimize any contention in operations between rooms.